boot/bootutil/include/bootutil/crypto/ecdh_p256.h

```diff
@@ -56,6 +64,72 @@ static inline int bootutil_ecdh_p256_shared_secret(bootutil_ecdh_p256_context *c
 }
 #endif /* MCUBOOT_USE_TINYCRYPT */
 
+#if defined(MCUBOOT_USE_MBED_TLS)
+typedef struct bootutil_ecdh_p256_context {
+    mbedtls_ecp_group grp;
+    mbedtls_ecp_point P;
+    mbedtls_mpi z;
+    mbedtls_mpi d;
+} bootutil_ecdh_p256_context;
+
+static inline void bootutil_ecdh_p256_init(bootutil_ecdh_p256_context *ctx)
+{
+    mbedtls_mpi_init(&ctx->z);
+    mbedtls_mpi_init(&ctx->d);
+
+    mbedtls_ecp_group_init( &ctx->grp );
+    mbedtls_ecp_point_init( &ctx->P );
+
+    if (0 != mbedtls_ecp_group_load( &ctx->grp, MBEDTLS_ECP_DP_SECP256R1 )) {
+        mbedtls_ecp_group_free( &ctx->grp );
+        mbedtls_ecp_point_free( &ctx->P );
```

Comment:
Don't need to free MPI's `z` and `d` here?

Reply:
z and d are only initialized with values by mbedtls_mpi_init(), no allocation